template-sample-app/template/src/components/views/Kpi.js

```diff
@@ -44,18 +43,27 @@ export default function Kpi() {
         transitionDuration: 500,
       })
     );
-
+    // Add the source query for the KPI
     dispatch(addSource(kpiSource));
-    dispatch(addLayer({ id: LAYER_ID, source: kpiSource.id }));
-    
+    // Add the layer
+    dispatch(
+      addLayer({
+        id: KPI_LAYER_ID,
+        source: kpiSource.id,
+        selectedStore: null,
+      })
+    );
+    // Close bottom panel
     dispatch(setBottomSheetOpen(false));
 
     return function cleanup() {
-      dispatch(removeLayer(LAYER_ID));
+      dispatch(removeLayer(KPI_LAYER_ID));
       dispatch(removeSource(kpiSource.id));
     };
   }, [dispatch]);
 
+  // Auto import useEffect
```

Comment:
Same. I guess this comment isn't required right? (I think I also saw another one somewhere to remove...)

Reply:
This is used to know where to place de useEffect import for add layers in a view.

Reply:
Sorry I was just thinking about the 1st use of the hygen. I wasn't thinking about a 2nd use and the need of this. Once you know it, it's pretty obvious it's required, but... are we explaining somewhere the need of these 'anchor' comments? If not, I think we must do it. 

And another suggestion to consider: if they are required for the tooling to work I would probably rename them, adding something more explicit, like

// hygen - Auto import useEffect - this line is required for hygen to work correctly

or something similar... to avoid the removal from the user (unless they know what they are doing)

What do you think guys?
(cc @albertoarana)

Reply:
That could be tackled in the next PR on hygen, if we eventually aggree on a change, so we can move this on

Reply:
Hi @VictorVelarde, I think that comment is mandatory because you can write code between useEffect and return code, so it's impossible that hyge works. I agree with you comment about extend the comment with `// hygen - Auto import useEffect - this line is required for hygen to work correctly

`

Reply:
I like more `// [hygen] Auto import useEffect - this line is required for hygen to work correctly`

Reply:
@albertoarana I don't like the `this line is required for hygen to work correctly`. The comment is pretty invasive just with `Auto import useEffect`. I'd just add `[hygen]` at the beginning as @albertoarana said and if you don't see it clear a comment in the documentation, but nothing else.

cc: @VictorVelarde

Reply:
I think that `// [hygen] Auto import useEffect` + a note in the docs about these comments may be enough for now